<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>DeepGenomeScan: A Deep Learning Approach for Whole Genome Scan (WGS) and Genome-wide Association Studies (GWAS) • DeepGenomeScan</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="DeepGenomeScan: A Deep Learning Approach for Whole Genome Scan (WGS) and Genome-wide Association Studies (GWAS)">
<meta property="og:description" content="This package implements the whole genome scan and genome-wide association studies using deep neural networks (i.e, Multi-Layer Perceptron (MLP), Convolutional Neural Network (CNN)). DeepGenomeScan offers heuristic learning and computational design integrating deep learning, robust resampling and cross validations methods, as well as Model-Agnostic interpretation of feature importance for convolutional neural networks. 
   DeepGenomeScan, in other words, deep learning for genome-wide scanning, is a deep learning approach for detecting variations under natural selection or omics-based association studies, such as GWAS, PWAS, TWAS, MWAS. 
   The framework makes the implemention user-friendly. Users can adopt the package's framework to study various ecological and evolutionary questions. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-158177382-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158177382-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">DeepGenomeScan</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/home.html">Home</a>
</li>
<li>
  <a href="articles/index.html">Tutorials</a>
</li>
<li>
  <a href="reference/index.html">Manual</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/xinghuq/DeepGenomeScan/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div id="deepgenomescan-a-deep-learning-approach-for-whole-genome-scan-wgs-and-genome-wide-association-studies-gwas" class="section level2">
<h2 class="hasAnchor">
<a href="#deepgenomescan-a-deep-learning-approach-for-whole-genome-scan-wgs-and-genome-wide-association-studies-gwas" class="anchor"></a>DeepGenomeScan : A Deep Learning Approach for Whole Genome Scan (WGS) and Genome-Wide Association Studies (GWAS)</h2>
<p>This package implements the genome scan and genome-wide association studies using deep neural networks (i.e, Multi-Layer Perceptron (MLP), Convolutional Neural Network (CNN)). DeepGenomeScan offers heuristic learning and computational design integrating deep learning (i.e.,Multi-Layer Perceptron (MLP), convolutional neural network(CNN)), robust resampling and cross validations methods, as well as Model-Agnostic interpretation of feature importance for convolutional neural networks. DeepGenomeScan, in other words, deep learning for genome-wide scanning, is a deep learning approach for detecting variations under natural selection or omics-based association studies, such as GWAS, PWAS, TWAS, MWAS. The package use the tensorflow as the backend. The design makes the implemention user-friendly. Users can adopt the package’s framework to study various ecological and evolutionary questions, but not only constraining in biology field.</p>
</div>
<div id="install-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#install-packages" class="anchor"></a>Install packages</h2>
<pre class="{r}"><code><a href="https://github.com/r-lib/devtools">
library("devtools")

devtools::install_github("xinghuq/DeepGenomeScan")

devtools::install_github("xinghuq/CaretPlus/pkg/caret")
</a></code></pre>
<div id="dependencies-and-environment-requirements" class="section level3">
<h3 class="hasAnchor">
<a href="#dependencies-and-environment-requirements" class="anchor"></a>Dependencies and environment requirements</h3>
<div id="note-environment-requirements-python-should-be-installed-and-the-python-package-of-keras-and-tensorflow-should-also-be-installed-and-work-properly-with-the-system" class="section level4">
<h4 class="hasAnchor">
<a href="#note-environment-requirements-python-should-be-installed-and-the-python-package-of-keras-and-tensorflow-should-also-be-installed-and-work-properly-with-the-system" class="anchor"></a>Note: Environment requirements: python should be installed and the python package of Keras and Tensorflow should also be installed and work properly with the system</h4>
<pre class="{r}"><code><a href="https://github.com/xinghuq/KLFDAPC">requireNamespace("KLFDAPC")

 if (!requireNamespace("KLFDAPC", quietly=TRUE))

  devtools::Install_github("xinghuq/KLFDAPC")
  
 if (!requireNamespace("DA", quietly=TRUE))
 
  devtools::install_github("xinghuq/DA")

if (!requireNamespace("keras", quietly=TRUE))

 Install.packages("keras")
  
 if (!requireNamespace("tensorflow", quietly=TRUE))
 
  install.packages("tensorflow")
 if (!requireNamespace("kerasR", quietly=TRUE))
 
  install.packages("kerasR")
</a></code></pre>
</div>
</div>
<div id="checking-the-python-environment" class="section level3">
<h3 class="hasAnchor">
<a href="#checking-the-python-environment" class="anchor"></a>Checking the python environment</h3>
<pre class="{r}"><code>
library("rappdirs")
library("reticulate")
reticulate::use_python("/usr/bin/python3")
library(caret) ### for ML calling functions and performance estimation, users should use the modified version at xinghuq/CaretPlus/caret instead of the original version
library(keras)  
library("tensorflow")

##checking if Tensorflow works properly
K0=keras::backend()
</code></pre>
</div>
<div id="library" class="section level3">
<h3 class="hasAnchor">
<a href="#library" class="anchor"></a>library</h3>
<pre class="{r}"><code><a href="https://github.com/xinghuq/DeepGenomeScan">library(DeepGenomeScan)
library(caret)### for ML calling functions and performance estimation
library(keras) ### for DL
library("tensorflow")
library("caretEnsemble")
library(kerasR)
library("RSNNS")
library(NeuralNetTools)
</a></code></pre>
</div>
<div id="example" class="section level3">
<h3 class="hasAnchor">
<a href="#example" class="anchor"></a>Example</h3>
</div>
<div id="preparing-data" class="section level3">
<h3 class="hasAnchor">
<a href="#preparing-data" class="anchor"></a>Preparing data</h3>
<pre class="{r}"><code>
f &lt;- system.file('extdata',package='DeepGenomeScan')
infile &lt;- file.path(f, "sim1.csv")
sim_example=read.csv(infile)
genotype=sim_example[,-c(1:14)]
env=sim_example[,2:11]
str(sim_example)
</code></pre>
</div>
<div id="setting-the-resampling-method" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-the-resampling-method" class="anchor"></a>Setting the resampling method</h3>
<pre class="{r}"><code>
econtrol1 &lt;- trainControl(## 5-fold CV, repeat 5 times
  method = "adaptive_cv",
  number = 5,
  ## repeated ten times
  repeats = 5,search = "random")
set.seed(999)
options(warn=-1
</code></pre>
</div>
<div id="deepgenomescan-with-mlp-model" class="section level3">
<h3 class="hasAnchor">
<a href="#deepgenomescan-with-mlp-model" class="anchor"></a>DeepGenomeScan with “mlp” model</h3>
<pre class="{r}"><code>
h2o_mlp&lt;- DeepGenomeScan(as.matrix(genotype_norm),env$envir1,
                                  method="mlp",
                                  metric = "RMSE",## "Accuracy", "RMSE","Rsquared","MAE"
                                  tuneLength = 10, ### 11 tunable parameters 11^2
                                  # tuneGrid=CNNGrid, ### or search 100 combinations of parameters using random tuneLength=100
                                  trControl = econtrol1)

#### varIMP for SNPs

out=varImp(h2o_mlp,scale = FALSE)
</code></pre>
<div id="plot-the-snp-importance-scores" class="section level5">
<h5 class="hasAnchor">
<a href="#plot-the-snp-importance-scores" class="anchor"></a>Plot the SNP importance scores</h5>
<pre class="{r}"><code><a href="https://rdrr.io/r/graphics/plot.html">
plot(out$Overall,  ylab="SNP importance")
</a></code></pre>
<p>Welcome any <a href="https://github.com/xinghuq/DeepGenomeScan/issues">feedback</a> and <a href="https://github.com/xinghuq/DeepGenomeScan/pulls">pull request</a>.</p>
</div>
</div>
<div id="version" class="section level3">
<h3 class="hasAnchor">
<a href="#version" class="anchor"></a>Version</h3>
<p>The current version is 0.5.5 (Sep 2, 2020).</p>
</div>
<div id="citation" class="section level3">
<h3 class="hasAnchor">
<a href="#citation" class="anchor"></a>Citation</h3>
<p>Qin, X. 2020. DeepGenomeScan. v0.5.5.</p>
<p>Qin, X., Gaggiotti, EO., Chiang, C. 2020. Detecting natural selection via deep learning. In submission.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/xinghuq/DeepGenomeScan/">https://​github.com/​xinghuq/​DeepGenomeScan/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/xinghuq/DeepGenomeScan/issues">https://​github.com/​xinghuq/​DeepGenomeScan/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 3)</li>
</ul>
</div>
<div class="citation">
<h2>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html">Citing DeepGenomeScan</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Xinghu Qin <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Xinghu Qin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
