<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Implementation of deep learning based genome scan
 — DeepGenomeScan • DeepGenomeScan</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Implementation of deep learning based genome scan
 — DeepGenomeScan" />

<meta property="og:description" content="Deep learning based genome scan with various ready-to-use models and self-defined models. 
" />





<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-158177382-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-158177382-1');
</script>


  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DeepGenomeScan</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../articles/home.html">Home</a>
</li>
<li>
  <a href="../articles/index.html">Tutorials</a>
</li>
<li>
  <a href="../reference/index.html">Manual</a>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xinghuq/DeepGenomeScan/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Implementation of deep learning based genome scan
<!-- %%  ~~function to do ... ~~ --></h1>
    
    <div class="hidden name"><code>DeepGenomeScan.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Deep learning based genome scan with various ready-to-use models and self-defined models. 
<!-- %%  ~~ A concise (1-5 lines) description of what the function does. ~~ --></p>
    
    </div>

    <pre class="usage">DeepGenomeScan(genotype, ...)

## Default S3 method:
DeepGenomeScan(
  genotype, 
  env,
  method = "mlp",
  preProcess = NULL,
  ...,
  weights = NULL,
  metric = ifelse(is.factor(y), "Accuracy", "RMSE"),
  maximize = ifelse(metric &lt;!-- %in% c("RMSE", "logLoss", "MAE"), FALSE, TRUE), --&gt;
  trControl = trainControl(),
  tuneGrid = NULL,
  tuneLength = ifelse(trControl$method == "none", 1, 3),
seed = 123)

## S3 method for class 'formula'
DeepGenomeScan(form, data, ..., weights, subset, na.action = na.fail, contrasts = NULL,seed = 123)

## S3 method for class 'recipe'
DeepGenomeScan(
  genotype,
  data,
  method = "mlp",
  ...,
  metric = ifelse(is.factor(y_dat), "Accuracy", "RMSE"),
  maximize = ifelse(metric &lt;!-- %in% c("RMSE", "logLoss", "MAE"), FALSE, TRUE), --&gt;
  trControl = trainControl(),
  tuneGrid = NULL,
  tuneLength = ifelse(trControl$method == "none", 1, 3),
seed = 123)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>genotype</th>
      <td><p>The genotype matrix, or other omics-based variations matrix. This is a  matrix where samples are in rows and variations/features are in columns. This could be a simple matrix, data frame or other type (e.g. sparse matrix) but must have column names (see Details below). Preprocessing using the preProcess argument only supports matrices or data frames. When using the recipe method, _"genotype"_ should be an unprepared recipe object that describes the model terms (i.e. outcome, predictors, etc., e.g. the h2o model in our example) as well as any pre-processing that should be done to the data. This is an alternative approach to specifying the model. Note that, when using the recipe method, any arguments passed to preProcess will be ignored. See the links and example below for more details using recipes.
<!-- %%     ~~Describe \code{x} here~~ --></p></td>
    </tr>
    <tr>
      <th>env</th>
      <td><p>Env/traits data. The environmental variables/geographic coordinates, or the latent geographic genetic structure vairbales produced from KLFDAPC. For GWAS, this shoud be the traits or phenotypic variables. The data can be a numeric or factor vector corresponding to each sample.
<!-- %%     ~~Describe \code{x} here~~ --></p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>The read-to-use models or user defined models. A string specifying which model to use. A list of functions can also be passed for a custom model function, see our example tutorials.
<!-- %%     ~~Describe \code{\dots} here~~ --></p></td>
    </tr>
    <tr>
      <th>...</th>
      <td><p>Arguments passed to the defined models (such as "mlpWeightDecayML"). Errors will occur if values for tuning parameters are passed here.
<!-- %%     ~~Describe \code{\dots} here~~ --></p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>Deep learning based genome scan with various ready-to-use models and self-defined models. Users can choose the available models in this package, and can also construct their own models then compile their model using DeepGenomeScan framework.
<!-- %%  ~~ If necessary, more details than the description above ~~ --></p>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p><!-- %%  ~Describe the value returned -->
<!-- %%  If it is a LIST, use --></p>
<dt>model_train</dt><dd><p>The optimal model trained from specific model and resampling methods</p></dd>

%% ...

    
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Abadi, M., Barham, P., Chen, J., Chen, Z., Davis, A., Dean, J., ... &amp; Kudlur, M. (2016). Tensorflow: A system for large-scale machine learning. In 12th USENIX symposium on operating systems design and implementation (OSDI 16) (pp. 265-283).</p>
<p>Bergmeir, C. N., &amp; Benítez Sánchez, J. M. (2012). Neural networks in R using the Stuttgart neural network simulator: RSNNS. American Statistical Association.</p>
<p>Beck, M. W. (2018). NeuralNetTools: Visualization and analysis tools for neural networks. Journal of statistical software, 85(11), 1.</p>
<p>Candel, A., Parmar, V., LeDell, E., &amp; Arora, A. (2016). Deep learning with H2O. H2O. ai Inc.</p>
<p>Deane-Mayer, Z. A., &amp; Knowles, J. E. (2016). caretEnsemble: ensembles of caret models. R package version, 2(0).</p>
<p>Gulli, A., &amp; Pal, S. (2017). Deep learning with Keras. Packt Publishing Ltd.</p>
<p>Klima, G. (2016). FCNN4R: Fast Compressed Neural Networks for R. R package version 0.6, 2.</p>
<p>Kuhn, M. (2015). Caret: classification and regression training. ascl, ascl-1505.</p>
<p>Qin, X. 2020. DeepGenomeScan. v0.5.5.</p>
<p>Qin, X., Gaggiotti, EO., Chiang, C. 2020. Detecting natural selection via deep learning. In submission.</p>
    
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>qinxinghu@gmail.com
<!-- %%  ~~who you are~~ --></p>
    
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>https://github.com/xinghuq/DeepGenomeScan/
<!-- %%  ~~further notes~~ --></p>
    

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Xinghu Qin.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.9000.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


